testsuite/cluster-test/src/experiments/state_sync_performance.rs

```diff
@@ -0,0 +1,169 @@
+    type E = StateSyncPerformance;
+
+    fn build(self, cluster: &Cluster) -> Self::E {
+        let validator_instance = cluster.random_validator_instance();
+        let fullnode_instance = cluster.random_fullnode_instance();
+        Self::E {
+            params: self,
+            fullnode_instance,
+            validator_instance,
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for StateSyncPerformance {
+    fn affected_validators(&self) -> HashSet<String> {
+        let mut result = HashSet::new();
+        result.insert(self.validator_instance.peer_name().clone());
+        result
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        // Stop the fullnode and clear all data so that it falls behind
+        info!("Stopping the fullnode: {}", self.fullnode_instance);
+        self.fullnode_instance.stop().await?;
+        self.fullnode_instance.clean_data().await?;
+
+        // Execute and commit transactions on the validators for the specified duration
+        let emit_transactions_duration_secs = self.params.emit_transactions_duration_secs;
+        info!(
+            "Executing transactions for {} seconds",
+            emit_transactions_duration_secs
+        );
+        let emit_job_request = EmitJobRequest::for_instances(
+            context.cluster.validator_instances().to_vec(),
+            context.global_emit_job_request,
+            0,
+            0,
+        );
+        let _ = context
+            .tx_emitter
+            .emit_txn_for(
+                Duration::from_secs(emit_transactions_duration_secs),
+                emit_job_request,
+            )
+            .await?;
+
+        // Read the validator synced version
+        let validator_synced_version = self.read_validator_synced_version();
+        info!(
+            "The validator is now synced at version: {}",
+            validator_synced_version
+        );
+
+        // Restart the fullnode so that it starts state syncing to catch up
+        info!(
+            "Waiting for the fullnode to wake up: {}",
+            self.fullnode_instance
+        );
+        self.fullnode_instance.start().await?;
+        self.fullnode_instance
+            .wait_json_rpc(Instant::now() + Duration::from_secs(120))
+            .await?;
+
+        // Wait for the fullnode to catch up to the expected version
+        info!(
+            "The fullnode is now up. Waiting for it to state sync to the expected version: {}",
+            validator_synced_version
+        );
+        let start_instant = Instant::now();
+        while self.read_fullnode_synced_version() < validator_synced_version {
+            time::sleep(Duration::from_secs(1)).await;
+        }
+        info!(
+            "The fullnode has caught up to version: {}",
+            validator_synced_version
+        );
+
+        // Calculate the state sync throughput
+        let time_to_state_sync = start_instant.elapsed();
```

Comment:
Maybe should call out the accuracy of the experiment is limited to second accuracy.  So, it's limited to the (# of versions) / ~(0-600).  Luckily I don't think it's going to be very fast that it'll matter :)

Reply:
Yeah, it's not great that `as_millis(..)` returns the number of whole seconds as milliseconds, as opposed to the number of whole seconds and any milliseconds: https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_millis

Either way, it's probably fine for now! We might make it more intelligent in the future :)